src/HttpClientRequestHandlerAmp.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace unreal4u\TelegramAPI;
+
+use Amp\Artax\DefaultClient;
+use Amp\Artax\Request;
+use Amp\Artax\Response;
+use Amp\Promise;
+use React\Promise\PromiseInterface;
+use unreal4u\TelegramAPI\Exceptions\ClientException;
+use unreal4u\TelegramAPI\InternalFunctionality\TelegramResponse;
+
+/**
+ * @param Promise $promise
+ * @return \React\Promise\PromiseInterface
+ */
+function reactAdapt(Promise $promise) {
+    $deferred = new \React\Promise\Deferred();
+
+    $promise->onResolve(function ($error = null, $result = null) use ($deferred) {
+        if ($error) {
+            $deferred->reject($error);
+        } else {
+            $deferred->resolve($result);
+        }
+    });
+
+    return $deferred->promise();
+}
+
+class HttpClientRequestHandlerAmp implements RequestHandlerInterface {
+
+    /**
+     * @var DefaultClient
+     */
+    private $client;
+
+    public function __construct() {
+        $this->client = new DefaultClient;
+    }
+
+    /**
+     * @param string $uri
+     *
+     * @return PromiseInterface
+     */
+    public function get(string $uri): PromiseInterface {
+         $request = (new Request($uri))->withMethod('GET');
```

Comment:
You don't need `->withMethod`, `GET` is used by default. For other methods, you can just use the second constructor argument.